tests/integration-tests/tests/integration/chained_authority_tests.rs

```diff
@@ -0,0 +1,363 @@
+use std::sync::Arc;
+
+use hickory_integration::TestResponseHandler;
+use hickory_proto::{
+    op::{Message, MessageType, Query, ResponseCode},
+    rr::{rdata::A, LowerName, Name, RData, Record, RecordSet, RecordType},
+    serialize::binary::{BinDecodable, BinEncodable},
+};
+use hickory_server::{
+    authority::{
+        Authority, Catalog, LookupControlFlow, LookupError, LookupObject, LookupOptions,
+        LookupRecords, MessageRequest, UpdateResult, ZoneType,
+    },
+    server::{Protocol, Request, RequestInfo, ResponseInfo},
+};
+
+use test_support::subscribe;
+
+#[cfg(feature = "dnssec")]
+use hickory_server::{authority::Nsec3QueryInfo, config::dnssec::NxProofKind};
```

Comment:
Nit: keep the `test_support` and conditional `hickory_server` imports in the same block as the other hickory imports?